src/dreyfus_index_updater.erl

```diff
@@ -31,6 +31,7 @@ update(IndexPid, Index) ->
     erlang:put(io_priority, {view_update, DbName, IndexName}),
     {ok, Db} = couch_db:open_int(DbName, []),
     try
+        {ok, ExcludeIdRevs} = purge_index(Db, IndexPid, Index),
```

Comment:
Hi @mayya-sharipova .
1. I used to delete the index first and update(rebuild) it in PR(#11). But Robert Newson asked me to `go from its current state to the new state without an intermediate state`, and `Here we will make the intermediate state, where the Documents in the Lucene index are removed, visible to search results.` So we delete or delete index instead of delete first and then update.
2. Based 1, we found that the index maybe delete/update twice, first in `purge_index` and second in `load_docs`. And update and delete in `clouseau` is expensive so we add an `ExcludeList` to avoid the unnecessary operations.